x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/action/TransportUpgradeTransformsAction.java

```diff
@@ -95,12 +96,12 @@ protected void masterOperation(Task ignoredTask, Request request, ClusterState s
         TransformNodes.warnIfNoTransformNodes(state);
 
         // do not allow in mixed clusters
-        if (state.nodes().getMaxNodeVersion().after(state.nodes().getMinNodeVersion())) {
+        if (TransportVersionUtils.isMinTransformVersionSameAsCurrent(state) == false) {
             listener.onFailure(
                 new ElasticsearchStatusException(
-                    "Cannot upgrade transforms. All nodes must be the same version [{}]",
+                    "Cannot upgrade transforms. All transport versions must be the same [{}]",
```

Comment:
nit, similar suggestion:

"Cannot upgrade transform while cluster upgrade is in progress".